slybot/slybot/spider.py

```diff
@@ -193,3 +222,41 @@ def handle_rss(self, response):
 
     def handle_html(self, response):
         return self._handle('handle_html', response)
+
+    def _build_js_url_filter(self, spec):
+        if not self.js_enabled:
+            return lambda x: None
+        enable_patterns = spec.get('js_enable_patterns')
+        disable_patterns = spec.get('js_disable_patterns')
+        filterf = None
+        enablef = None
+        if enable_patterns:
+            pattern = enable_patterns[0] if len(enable_patterns) == 1 else \
+                "(?:%s)" % '|'.join(enable_patterns)
+            pattern_re = re.compile(pattern)
+            enablef = lambda x: pattern_re.search(x)
+        if disable_patterns:
+            pattern = disable_patterns[0] if len(disable_patterns) == 1 else \
+                "(?:%s)" % '|'.join(disable_patterns)
+            pattern_re = re.compile(pattern)
+            disablef = lambda x: pattern_re.search(x)
+            if not enablef:
+                filterf = lambda x: not disablef(x)
+            else:
+                filterf = lambda x: enablef(x) and not disablef(x)
+        return filterf if filterf else lambda x: x
```

Comment:
In case there is only `enable_patterns` and no `disable_patterns` shouldn't it accept it if any of `enable_patterns` matches?
because now if there are no disable_patterns it will accept any url.
```
_build_js_url_filter({
    "js_enable_patterns": ['foo', 'bar']
})('hello world')
>>> 'hello world'
_build_js_url_filter({
    "js_enable_patterns": ['foo', 'bar'],
    "js_disable_patterns": ['baz']
})('hello world')
>>> None
```